wrappers/python/simtk/openmm/app/modeller.py

```diff
@@ -1451,20 +1451,80 @@ def addMembrane(self, forcefield, lipidType='POPC', membraneCenterZ=0*nanometer,
             numPositive += numPairs
             numNegative += numPairs
         totalIons = numPositive+numNegative
-        
-        # Now add the ions.  We do this by randomly selecting a set of waters that were just added and replacing
-        # them with ions.
 
+        # Now add the ions. We do this by randomly selecting a set of waters a certain
+        # distance away from either leaflet of the membrane and replacing them with ions,
+        # while ensuring ions are not placed too close to each other.
         if totalIons > 0:
-            toReplace = random.sample(list(waterPos), totalIons)
+
+            # Redo waterPos because previous modeller.delete() changes Chain indexes
+            waterPos = {}
+            for chain in list(modeller.topology.chains())[-2:]:
+                for residue in chain.residues():
+                    if residue.name == 'HOH':
+                        for atom in residue.atoms():
+                            if atom.element == elem.oxygen:
+                                waterPos[residue] = modeller.positions[atom.index].value_in_unit(nanometer)
+
+            # Calculate lipid Z boundaries
+            lipidNames = {res.name for res in patch.topology.residues() if res.name != 'HOH'}
+            lipidZMax = 0.0
+            lipidZMin = 99999.0
```

Comment:
Use `sys.float_info.min` and `sys.float_info.max`.  Users are free to pass in their own patch, and we can't assume what range the positions will be in.  (Ok, assuming they'll be less than 99999 is pretty safe.  But we shouldn't assume they'll be positive.)

Reply:
Indeed, changed!